src/nodewatcher/plugins/slurm.py

```diff
@@ -54,44 +61,28 @@ def has_pending_jobs(instance_properties, max_size):
 
 
 def lock_host(hostname, unlock=False):
-    # hostname format: ip-10-0-0-114.eu-west-1.compute.internal
-    hostname = hostname.split(".")[0]
-    if unlock:
-        log.info("Unlocking host %s", hostname)
-        command = [
-            "/opt/slurm/bin/scontrol",
-            "update",
-            "NodeName={0}".format(hostname),
-            "State=RESUME",
-            'Reason="Unlocking"',
-        ]
+    if unlock_node:
+        unlock_node(hostname)
     else:
-        log.info("Locking host %s", hostname)
-        command = [
-            "/opt/slurm/bin/scontrol",
-            "update",
-            "NodeName={0}".format(hostname),
-            "State=DRAIN",
-            'Reason="Shutting down"',
-        ]
-    try:
-        run_command(command)
-    except subprocess.CalledProcessError:
```

Comment:
This was previously swallowing the exception in case of failure. Now we are not. Let's make sure we properly handle the error cases

Reply:
Preserving old behavior, locking action now will not raise error, `sqswatcher` and `nodewatcher` can handle locking failures on their own without catching the error